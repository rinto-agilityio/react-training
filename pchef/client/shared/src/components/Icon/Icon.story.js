// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Themes
import { COLORS } from '../../themes'

// Components
import Icon from '.'

// Helper
import { wInfo } from '../../../.storybook/utils';

storiesOf('Icon', module)
  .addDecorator(wInfo())
  .add('Default', () => (
    <Icon
      onPress={action('clicked')}
      color={COLORS.icon.default}
      name="add-a-photo"
    />
  ))
  .add('Disabled', () => (
    <Icon
      onPress={action('clicked')}
      color={COLORS.icon.default}
      name="add-a-photo"
      disabled
    />
  ))
