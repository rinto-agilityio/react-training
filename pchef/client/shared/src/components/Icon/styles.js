// Lib
import { StyleSheet } from 'react-native'

// Themes
import { METRICS } from '../../themes'

// Create style for Icon
const styles = StyleSheet.create({
  wrapIcon: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  label: {
    marginLeft: METRICS.mediumMargin,
    textAlign: 'center',
  },
})

export default styles
