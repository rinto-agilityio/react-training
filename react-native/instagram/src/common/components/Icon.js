// Libs
import React from 'react'
import { Image } from 'react-native'
import PropTypes from 'prop-types'

const Icon = ({ style, source }) => <Image style={style} source={source} />

Icon.propTypes = {
  style: PropTypes.any,
  source: PropTypes.any
}

export default Icon
