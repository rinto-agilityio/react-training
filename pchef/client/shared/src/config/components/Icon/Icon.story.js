// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Themes
import { COLORS } from '../../themes'

// Components
import Icon from '.'

storiesOf('Icon', module)
  .add('default', () => (
    <Icon
      onPress={action('clicked')}
      color={COLORS.baseBlue}
      name="add-a-photo"
    />
  ))
  .add('disabled', () => (
    <Icon
      onPress={action('clicked')}
      color={COLORS.baseBlue}
      name="add-a-photo"
      disabled
    />
  ))
