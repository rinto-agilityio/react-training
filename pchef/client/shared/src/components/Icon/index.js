// Libs
import React from 'react'
import { View, Text } from 'react-native'
import { IconButton } from 'react-native-paper'

// Styles
import styles from './styles'

// Themes
import { COLORS, FONTS } from '../../themes'

type Props = {
  label?: string | number,
  onPress?: () => void | Promise<void>,
  customStyle?: {} | Array<{}>,
  color?: string,
  size?: number,
  name: string,
  disabled?: boolean,
  wrapperIconStyle?: Object,
  customIconStyles?: {},
};

const Icon = ({
  label,
  onPress,
  customStyle,
  color,
  size,
  name,
  wrapperIconStyle,
  disabled,
  customIconStyles,
}: Props) => (
  <View style={[
    styles.wrapIcon,
    wrapperIconStyle,
  ]}
  >
    <IconButton
      icon={name}
      color={color}
      size={size}
      onPress={onPress}
      disabled={disabled}
      style={[styles.iconStyles, customIconStyles]}
    />
    {label ? <Text style={[styles.label, customStyle]}>{label}</Text> : null}
  </View>
)

Icon.defaultProps = {
  label: '',
  onPress: () => {},
  customStyle: {},
  size: FONTS.fontSize.medium,
  color: COLORS.black,
  wrapperIconStyle: {},
  disabled: false,
  customIconStyles: {},
};

export default Icon
