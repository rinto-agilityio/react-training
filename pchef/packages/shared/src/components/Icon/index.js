// libs
import React from 'react';
import { View, Text } from 'react-native';
import { Icon as IconComponent } from 'react-native-elements'

// Styles
import styles from './styles';

// Themes
import { COLORS, METRICS } from '../../themes';

type Props = {
  label?: string,
  onClick?: () => void,
  customStyle?: {},
  color?: string,
  type?: string,
  size?: number,
  name: string
};

const Icon = ({
  label,
  onClick,
  customStyle,
  color,
  type,
  size,
  name
}: Props) => (
  <View style={styles.wrapIcon}>
    <IconComponent
      name={name}
      type={type}
      color={color}
      size={size}
      onPress={onClick}
    />
    {label ? <Text style={[customStyle, styles.label]}>{label}</Text> : null}
  </View>
);

export default Icon;

Icon.defaultProps = {
  label: '',
  type: 'font-awesome',
  onClick: () => {},
  customStyle: {},
  size: METRICS.fontSize.medium,
  color: COLORS.black
};
