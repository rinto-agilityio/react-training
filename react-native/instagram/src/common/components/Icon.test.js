// Components
import Icon from './Icon'

describe('Icon component', () => {
  const iconHome = require('@assets/icons/home.png')

  it('Renders correctly', () => {
    const treeDOM = renderer.create(<Icon source={iconHome} />).toJSON()

    expect(treeDOM).toMatchSnapshot()
  })
})
