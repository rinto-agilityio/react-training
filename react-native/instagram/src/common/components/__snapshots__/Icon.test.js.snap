// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component Renders correctly 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/icons/home.png",
    }
  }
  style={undefined}
/>
`;
